### PR TITLE
Make use of GitHub templates

### PR DESCRIPTION
**Pull request by Grinch**
**Changes made within this pull request**
 - [x] Added PULL_REQUEST_TEMPLATE.md
 - [x] Added ISSUE_TEMPLATE.md
 - [x] Moved CONTRIBUTING.md to .github/

Examples: [Pull Request template](https://gist.github.com/Grinch/4a6b52905a63c0a10058) and [Issue template](https://gist.github.com/Grinch/7fba2e0ee012e677ac8b)

Signed-off-by: Steven Downer <grinch@outlook.com>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!-- Be as accurate as possible. Add/remove as needed to keep things informative but also relevant. -->
+**I am currently running...**
+* Implementation
```
Comment (Aaron1011) [inline .github/ISSUE_TEMPLATE.md]: Shouldn't we have a separate template for SpongeForge and SpongeVanilla?

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ - [ ] SpongeForge
+ - [ ] SpongeVanilla
+* Implementation version:
```
Comment (Aaron1011) [inline .github/ISSUE_TEMPLATE.md]: There should be a 'Forge Version' field too.

**Comment by Grinch** [comment]
@Aaron1011 Updated. That ISSUE_TEMPLATE wasn't meant for this repo.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<!-- Be as accurate as possible. Add/remove as needed to keep things informative but also relevant. -->
+**I am currently running...**
+* SpongeForge version:
```
Comment (kashike) [inline .github/ISSUE_TEMPLATE.md]: Minecraft Forge version would be nice

**Comment by Grinch** [comment]
@kashike Done.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<!-- Delete any related links that are not relevant or applicable to this PR. -->
+**Related:** [SpongeAPI](REPLACE_WITH_LINK) | [SpongeCommon](REPLACE_WITH_LINK)
```
Comment (liach) [inline .github/PULL_REQUEST_TEMPLATE.md]: I suggest to make it into:
````md
[SpongeAPI][1] | [SpongeCommon][2] | **SpongeForge** | [SpongeVanilla][3]

[1]: https://github.com/SpongePowered/SpongeAPI/pull/<PR#>
[2]: https://github.com/SpongePowered/SpongeCommon/pull/<PR#>
[3]: https://github.com/SpongePowered/SpongeVanilla/pull/<PR#>
````

**Pull request closed without merging**